src/librustc_mir/util/liveness.rs

```diff
@@ -455,3 +544,42 @@ pub fn write_mir_fn<'a, 'tcx>(
     writeln!(w, "}}")?;
     Ok(())
 }
+
+crate struct NllLivenessMap {
```

Comment:
I would expect this to live in the nll code -- I'd probably put it in a new module `nll/liveness_map.rs` along with the `LocalWithRegion` index declaration.